dali/base/dadfs.cpp

```diff
@@ -177,21 +177,27 @@ static IPropertyTree *getEmptyAttr()
     return createPTree("Attr");
 }
 
-static double calcFileCost(const char * cluster, double sizeGB, double fileAgeDays)
+static double calcFileCost(const char * cluster, double sizeGB, double fileAgeDays, __int64 numDiskWrites, __int64 numDiskReads)
 {
     Owned<IPropertyTree> plane = getStoragePlane(cluster);
-    double atRestCost = 0.0;
+    Owned<IPropertyTree> global;
+    IPropertyTree * costPT = nullptr;
+
     if (plane && plane->hasProp("cost/@storageAtRest"))
     {
-        atRestCost = plane->getPropReal("cost/@storageAtRest", 0.0);
+        costPT = plane->queryPropTree("cost");
     }
     else
     {
-        Owned<IPropertyTree> global = getGlobalConfig();
-        atRestCost = global->getPropReal("cost/@storageAtRest", 0.0);
+        global.setown(getGlobalConfig());
+        costPT = global->queryPropTree("cost");
     }
-    double storageCostDaily = atRestCost * 12 / 365;
-    return storageCostDaily * sizeGB * fileAgeDays;
+    constexpr int accessPriceScalingFactor = 10000; // read/write pricing based on 10,000 operations
+    double atRestPrice = costPT->getPropReal("@storageAtRest", 0.0);
+    double readPrice =  costPT->getPropReal("@storageReads", 0.0);
+    double writePrice =  costPT->getPropReal("@storageWrites", 0.0);
+    double storageCostDaily = atRestPrice * 12 / 365;
+    return (storageCostDaily * sizeGB * fileAgeDays) + (readPrice * numDiskReads / accessPriceScalingFactor) + (writePrice * numDiskWrites / accessPriceScalingFactor);
```

Comment:
trivial: needed to check that the multiply would be done 1st, before the divide, otherwise it would lose precision.  Possibly better would be a double, or a value for multiplication.
double accessScalingFactor = 0.0000001;
correct, as it stands though.

Reply:
The order of precedence has multiplication and division same level and from left to right: https://en.cppreference.com/w/cpp/language/operator_precedence.   Compilers shouldn't re-order a multiply and divisions for the very reason you've mentioned.

Reply:
Yes, that's why I said it was correct - but I had to double check the spec to be sure.